composer/core/state.py

```diff
@@ -651,6 +651,55 @@ def get_elapsed_duration(self) -> Optional[Time[float]]:
             return None
         return self.timestamp.get(self.max_duration.unit) / self.max_duration
 
+    def convert_time(self, time: Union[str, Time[int], Time[float]], ssr: float = 1.0) -> Time[int]:
+        """Converts ``time`` into a fine-grained :class:`.TimeUnit` according to the current state.
+
+        It will not convert ``TimeUnit.BATCH``, ``TimeUnit.SAMPLE`` and ``TimeUnit.TOKEN``.
+        ``TimeUnit.Epoch`` is converted to ``TimeUnit.BATCH```.
+        Finally, ``TimeUnit.DURATION`` is converted either to ``TimeUnit.BATCH`` or ``TimeUnit.SAMPLE`` if ``.max_duration`` is ``TimeUnit.SAMPLE``.
+
+        .. note::
+
+            ``.max_duration`` cannot be `None`.
+
+            ``.dataloader_len`` cannot be `None` if,
+                * ``time.unit`` is ``TimeUnit.EPOCH``
+                * ``time.unit`` is ``TimeUnit.DURATION`` and ``.max_duration.unit`` is ``TimeUnit.EPOCH``.
+
+            Scale Schedule Ratio (ssr) is not applied if ``time.unit`` is ``TimeUnit.DURATION``
+
+        Args:
+            time (Time | str): A time string, or instance of :class:`.Time`.
+            ssr (float): Scale Schedule Ratio.
+
+        Returns:
+            Time: An instance of :class:`.Time`.
+        """
+        if isinstance(time, str):
+            time = Time.from_timestring(time)
+
+        if self.max_duration is None:
+            raise RuntimeError('`max_duration` should be set whenever time conversion is invoked.')
+
+        if time.unit == TimeUnit.DURATION:
+            if self.max_duration.unit == TimeUnit.EPOCH:
+                if self.dataloader_len is None:
+                    raise RuntimeError('Cannot convert time, as state.dataloader_len is None.')
+                return Time(
+                    int(time.value * int(self.dataloader_len) * self.max_duration.value),
```

Comment:
@mvpatel2000, I am not sure if we should apply `ssr` here. I decided to keep the original implementation but I do not see any reason to apply this scaling once the TimeUnit has changed to BATCH. Similarly to the `elif` case.

Reply:
No worries, feel free to edit the PR 😄 

Thanks for your efforts. I am happy to keep contributing with PRs or Issues if I find something.

Let me know if I can help.

Reply:
@priba apologies for the delay, this might be on the backburner for a few weeks. Really slammed with other things. If this is a blocker, please let me know and we can try landing a smaller solution. However, I'd really like to get to the full solution and make this much easier for everyone.

Reply:
No problem, we already have a workaround. Although it would be a nice feature, I understand it is not a high priority :)

Reply:
Hi @mvpatel2000 , are there any news regarding this PR?

Reply:
Sorry, I wasn't able to get to it yet. I will make sure to prioritize over next 2-3 weeks!

Reply:
@priba  sorry for delay, I still haven't gotten to it. I will make sure to do in January though. I keep getting pre-empted with higher priority items